ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -0,0 +1,154 @@
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use anyhow::Result;
+use axum::{
+    body::boxed,
+    http::{header, Request, Uri},
+    response::{IntoResponse, Response},
+    Json,
+};
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+use tabby_common::{config::RepositoryConfig, SourceFile, Tag};
+use tokio::sync::OnceCell;
+use tower::ServiceExt;
+use tower_http::services::ServeDir;
+use tracing::error;
+
+static DATASET: OnceCell<HashMap<DatasetKey, Meta>> = OnceCell::const_new();
+
+const MIME_VENDOR: &str = "application/vnd.directory+json";
+
+#[derive(Hash, PartialEq, Eq, Debug)]
+pub struct DatasetKey {
+    local_name: String,
+    rel_path: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Repository {
+    name: String,
+    path: Option<String>,
+}
+
+impl Repository {
+    pub fn dataset_key(&self) -> DatasetKey {
+        DatasetKey {
+            local_name: self.name.clone(),
+            rel_path: self.path_str().to_string(),
+        }
+    }
+
+    pub fn name_str(&self) -> &str {
+        self.name.as_str()
+    }
+
+    pub fn path_str(&self) -> &str {
+        self.path.as_deref().unwrap_or("")
+    }
+}
+
+#[derive(Serialize)]
+struct ListDir {
+    entries: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Meta {
+    git_url: String,
+    filepath: String,
+    language: String,
+    max_line_length: usize,
+    avg_line_length: f32,
+    alphanum_fraction: f32,
+    tags: Vec<Tag>,
+}
+
+impl From<SourceFile> for Meta {
+    fn from(file: SourceFile) -> Self {
+        Self {
+            git_url: file.git_url,
+            filepath: file.filepath,
+            language: file.language,
+            max_line_length: file.max_line_length,
+            avg_line_length: file.avg_line_length,
+            alphanum_fraction: file.alphanum_fraction,
+            tags: file.tags,
+        }
+    }
+}
+
+/// Load dataset
+pub async fn load_dataset() -> Result<()> {
```

Comment:
```suggestion
async fn load_meta() -> Result<()> {
```

This function is no longer needed to be public, and the static variable can be initialized with `lazy_static`

Reply:
`load_meta` is used in `repositories.rs` file which requires it to be `public`, or compiler is unhappy.

I narrow down it to `pub(crate)` anyway.